Bootloader/Bootloader/Mcal/Adc/Adc.c

```diff
@@ -1,6 +1,222 @@
+* FILE INFORMATION
+* -------------------------------------------------------------------------------------------------
+* NAME:              Adc.c
+* MODULE:            Adc
+* AUTHOR:            ShroukRashwan
+* DESCRIPTION:       Adc driver functional implementation
+* CREATION DATE:     7/2/2020
+* MODIFICATION DATE: 12/2/2020
+**************************************************************************************************/
+
+/**************************************************************************************************
+*                                       INCLUDES & LIBRARIES                                      *
+**************************************************************************************************/
+#include "Adc.h"
+#include "Adc_Internal.h"
+
+/**************************************************************************************************
+*                                         GLOBAL VARIABLES                                        *
+**************************************************************************************************/
+#if UNIT_TESTING == STD_ON
+	uint8 Adc_ADMUX																				=0;
+	uint8 Adc_ADCSRA																			=0;
+	uint8 Adc_ADCH																				=0;
+	uint8 Adc_ADCL																				=0;
+	uint8 Adc_SFIOR																				=0;
+#endif
+
+uint16 Adc_Value 																				=0;
+
+
+/**************************************************************************************************
+*                                     FUNCTIONS IMPLEMENTATION                                    *
+**************************************************************************************************/
+boolean Adc_Init(uint8 Param_ChannelNumber, uint8 Param_VolageReference, uint8 Param_PreScaler)
+{
+	if (Param_ChannelNumber>=0 && Param_ChannelNumber <=7 && Param_VolageReference >=0 &&
+		Param_VolageReference <=3 &&	Param_PreScaler  >=0 && Param_PreScaler <= 7)
+	{
+		/* No Gain !! */
+		CLEAR_BIT(ADC_ADMUX,4);
+		CLEAR_BIT(ADC_ADMUX,3);
+		/* Choose channel according to user input */
+		switch(Param_ChannelNumber)
+		{
+			case 0:
+				CLEAR_BIT(ADC_ADMUX,2);
+				CLEAR_BIT(ADC_ADMUX,1);
+				CLEAR_BIT(ADC_ADMUX,0);
+			break;			
+			case 1:
+				CLEAR_BIT(ADC_ADMUX,2);
+				CLEAR_BIT(ADC_ADMUX,1);
+				SET_BIT(ADC_ADMUX,0);
+			break;			
+			case 2:
+				CLEAR_BIT(ADC_ADMUX,2);
+				SET_BIT(ADC_ADMUX,1);
+				CLEAR_BIT(ADC_ADMUX,0);
+			break;			
+			case 3:
+				CLEAR_BIT(ADC_ADMUX,2);
+				SET_BIT(ADC_ADMUX,1);
+				SET_BIT(ADC_ADMUX,0);
+			break;			
+			case 4:
+				SET_BIT(ADC_ADMUX,2);
+				CLEAR_BIT(ADC_ADMUX,1);
+				CLEAR_BIT(ADC_ADMUX,0);
+			break;			
+			case 5:
+				SET_BIT(ADC_ADMUX,2);
+				CLEAR_BIT(ADC_ADMUX,1);
+				SET_BIT(ADC_ADMUX,0);
+			break;			
+			case 6:
+				SET_BIT(ADC_ADMUX,2);
+				SET_BIT(ADC_ADMUX,1);
+				CLEAR_BIT(ADC_ADMUX,0);
+			break;			
+			case 7:
+				SET_BIT(ADC_ADMUX,2);
+				SET_BIT(ADC_ADMUX,1);
+				SET_BIT(ADC_ADMUX,0);
```

Comment:
I think all this switch case can be mapped to only one line of code, right? Think about it ;)